Add protections for module files that don't have exports

This should return some kind of a "Failed to load file" message or something instead of the current, unhandled error

```
node --enable-source-maps ./target/core/main.js 
|24/08/2023, 18:47:36| [II] core: Starting...
|24/08/2023, 18:47:37| [II] core: Initialized Discord connection
node:events:492
      throw er; // Unhandled 'error' event
      ^

TypeError: Cannot read properties of undefined (reading 'name')
    at generateSlashCommandForModule (C:\Users\Meti\Documents\TuringBot\src\core\slash_commands.ts:101:21)
    at Client.<anonymous> (C:\Users\Meti\Documents\TuringBot\src\core\main.ts:45:27)
Emitted 'error' event on Client instance at:
    at emitUnhandledRejectionOrErr (node:events:395:10)
    at process.processTicksAndRejections (node:internal/process/task_queues:84:21)

Node.js v18.17.0
make: *** [Makefile:22: start] Error 1```

--- a/src/core/main.ts
+++ b/src/core/main.ts
@@ -104,6 +104,18 @@ async function importModulesFromFile(path: string): Promise<void> {
   // get the *default* export from each file and add it to the array of modules
   // dynamic imports import relative to the path of the file being run
   const fileExport = await import(path);
+
+  if (fileExport.default === undefined) {
+    const fileName = path.split('/').slice(-1);
+    logEvent(
+      EventCategory.Warning,
+      'core',
+      `File '${fileName}' has no module exports, skipping it`,
+      1
+    );
+    return;
+  }
+
   // to allow multiple module exports from the same file, if they exported an array, then iterate over it
   if (Array.isArray(fileExport.default)) {
     for (const module of fileExport.default) {

